src/cpg_flow/stage.py

```diff
@@ -76,11 +76,24 @@ def __init__(
         stage: Optional['Stage'] = None,
     ):
         # Converting str into Path objects.
-        self.data: dict[str, Path] | Path | None = None
+        self.data: ExpectedResultT = None
         if isinstance(data, str):
```

Comment:
This might not be practical at this point, but:

As a general principle, it's probably better to not guess whether a string should be a path or not. Either the caller should pass an object of the right type (because it knows which values are paths), or the receiver should interpret the value in the context that it intends to use it (because it knows it's about to treat a value as a path).

Reply:
+1 from me, I hate this Compound type. The issue is that our current workflows have both data types, and whilst a workflow will be internally consistent it's not realistic for the workflow framework to assume how a workflow has been implemented. 

If this became a fixed convention (`expected_outputs` only returns Paths or collections of Paths, and all Paths will be checked for existence when building the DAG) that would probably add a lot of clarity, but doing that now will break some existing workflows.